server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorCleanupOvershadowed.java

```diff
@@ -82,6 +83,19 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
             stats.addToGlobalStat("overShadowedCount", 1);
           }
         }
+
+        for (LoadQueuePeon loadQueue : coordinator.getLoadManagementPeons().values()) {
```

Comment:
Can you simply add the loadQueues to the timeline before the immediately prior loop, and let the loop which looks for overshadowed segments find them?